pandas/io/parsers/arrow_parser_wrapper.py

```diff
@@ -91,20 +91,35 @@ def _finalize_output(self, frame: DataFrame) -> DataFrame:
             The processed DataFrame.
         """
         num_cols = len(frame.columns)
+        multi_index_named = True
         if self.header is None:
             if self.names is None:
                 if self.prefix is not None:
                     self.names = [f"{self.prefix}{i}" for i in range(num_cols)]
                 elif self.header is None:
                     self.names = range(num_cols)
+            if len(self.names) != num_cols:
+                # usecols is passed through to pyarrow, we only handle index col here
+                # The only way self.names is not the same length as number of cols is
+                # if we have int index_col. We should just pad the names(they will get
+                # removed anyways) to expected length then.
+                self.names = list(range(num_cols - len(self.names))) + self.names
+                multi_index_named = False
             frame.columns = self.names
         # we only need the frame not the names
         frame.columns, frame = self._do_date_conversions(frame.columns, frame)
         if self.index_col is not None:
             for i, item in enumerate(self.index_col):
                 if is_integer(item):
                     self.index_col[i] = frame.columns[item]
+                else:
+                    # String case
+                    if item not in frame.columns:
+                        raise ValueError(f"Index {item} invalid")
```

Comment:
hit in tests? is this an error which is common across parsers? should this be a KeyError? (this is basically an invalid index_cocl right?

Reply:
Yup, idk why it's a ValueError, though. It's hit by tests though.
https://github.com/pandas-dev/pandas/pull/43706/files#diff-6bbd84495202b9fe8941c1bdfaa4ac1077735e8334abf809b2830827fdd6dd4eR42-R45